### PR TITLE
style: DRY how addons are disabled on Azure Stack

### PR DESCRIPTION
**Pull request by jadarsie**
**Reason for Change**:
Addressing feedback from a previous PR regarding code organization.

**Issue Fixed**:
Related to issue #1224

**Requirements**:

**Notes**:

### CODE REVIEW

File: pkg/api/types.go [hunk 1]
```diff
@@ -1829,3 +1811,7 @@ func FormatProdFQDNByLocation(fqdnPrefix string, location string, cloudName stri
 	FQDNFormat := AzureCloudSpecEnvMap[targetEnv].EndpointConfig.ResourceManagerVMDNSSuffix
 	return fmt.Sprintf("%s.%s."+FQDNFormat, fqdnPrefix, location)
 }
+
+func (p *Properties) isKubernetesAndNotAzureStack() bool {
+	return p.OrchestratorProfile != nil && p.OrchestratorProfile.KubernetesConfig != nil && !p.IsAzureStackCloud()
```
Comment (CecileRobertMichon) [inline pkg/api/types.go]: I assume this is already unit tested through the other functions' unit tests?

Comment (jadarsie) [reply pkg/api/types.go]: We should see failed tests if this function is not doing what is supposed to do

Comment (jackfrancis) [reply pkg/api/types.go]: Is there a reason not to just make a method with a simpler name like "`isAzureStack()`" ? Then we can employ `!isAzureStack()` if we want to validate the NOT condition.

(Also as AKS-Engine + Azure Stack is very unlikely to ever be anything other than Kubernetes, I think we can hide the "is kubernetes" logic behind the method, but we don't need to advertise that in the method name)

Comment (mboersma) [reply pkg/api/types.go]: I had suggested this explicit function name, but `isAzureStack` probably covers it.

Comment (CecileRobertMichon) [reply pkg/api/types.go]: We want to check for non-nil orcherstrator profile values regardless of Azure Stack so I don't think we should embed that in the IsAzureStack func. If later we want to re-enable some or all of these addons for Azure Stack, we will still want to keep the `isKubernetes` validation.

Comment (jadarsie) [reply pkg/api/types.go]: Renamed function and kept nil checks.

File: pkg/api/types.go [hunk 2]
```diff
@@ -1829,3 +1811,7 @@ func FormatProdFQDNByLocation(fqdnPrefix string, location string, cloudName stri
 	FQDNFormat := AzureCloudSpecEnvMap[targetEnv].EndpointConfig.ResourceManagerVMDNSSuffix
 	return fmt.Sprintf("%s.%s."+FQDNFormat, fqdnPrefix, location)
 }
+
+func (p *Properties) isAzureStack() bool {
+	return p.OrchestratorProfile != nil && p.OrchestratorProfile.KubernetesConfig != nil && p.IsAzureStackCloud()
```
Comment (jackfrancis) [inline pkg/api/types.go]: I'm actually a little confused by this method now. What is the purpose of checking:

- non-nil OrchestratorProfile
- non-nil KubernetesConfig
- Azure Stack cloud definition

all together?

Why not just invoke the existing `p.IsAzureStackCloud()` method in the addons enabled methods?

Comment (jackfrancis) [reply pkg/api/types.go]: I see what's going on. Sigh. These "is addons enabled" methods should not be a method of the `Properties` object. Stand by.

Comment (CecileRobertMichon) [reply pkg/api/types.go]: they weren't. That was to accommodate using the `isAzureStackCloud` func in them.

Comment (jackfrancis) [reply pkg/api/types.go]: I'll make these changes in this PR: https://github.com/Azure/aks-engine/pull/1265

Comment (CecileRobertMichon) [reply pkg/api/types.go]: this PR is follow up from #1233

Comment (jackfrancis) [reply pkg/api/types.go]: @jadarsie see this commit:

https://github.com/Azure/aks-engine/pull/1265/commits/76411647e150d09487c412afde88a784846d454f

Basically, we should be gating addons in Azure Stack scenarios during the customData (cloud-init) string concatenation process, not in the "is this addon enabled?" method itself. That latter method we want to keep nice and lean, basically: "do I have this addon configured and enabled in my api model?".

Comment (jackfrancis) [reply pkg/api/types.go]: Also, as @CecileRobertMichon pointed out, I should have been more opinionated in the prior PR.

Comment (jadarsie) [reply pkg/api/types.go]: Sounds good @jackfrancis, just let me know your preferences.

The changes look OK, it looks similar to my first commit: https://github.com/Azure/aks-engine/pull/1233/commits/b8e70377ed7ebe403bd01ab82cab2da0b5691994

Comment (CecileRobertMichon) [reply pkg/api/types.go]: I think the reason we moved the `!isAzureStack()` condition one level up was to be able to unit test it.

**Comment by jadarsie** [comment]
Fixed by #1265

**Pull request closed without merging**